interop-cats/shared/src/main/scala/zio/stream/interop/FS2StreamSyntax.scala

```diff
@@ -20,7 +20,7 @@ class ZStreamSyntax[R, E, A](private val stream: ZStream[R, E, A]) extends AnyVa
 
   /** Convert a [[zio.stream.ZStream]] into an [[fs2.Stream]]. */
   def toFs2Stream: fs2.Stream[ZIO[R, E, *], A] =
-    fs2.Stream.resource(stream.process.toResourceZIO).flatMap { pull =>
+    fs2.Stream.resource[ZIO[R, E, *], ZIO[R, Option[E], Chunk[A]]](stream.process.toResourceZIO).flatMap { pull =>
```

Comment:
Another Scala 3 issue?

Reply:
Yes 👍

Reply:
I hope we are capturing these somewhere.

Reply:
The problem is that we have `Async[RIO[Clock & Blocking, *]]` and `GenConcurrent[ZIO[R, E, *], E]` and here we need `MonadCancel[F, _]`. `Async` takes priority because it's a subtype. But somehow Scala 2 does a better job and picks the `Concurrent` instance. `compile` has the same problem.

Reply:
I think the expected result type `fs2.Stream[ZIO[R, E, *], A]` should be taken into account by type inference.